perfschema/statement.go

```diff
@@ -160,7 +160,10 @@ func (ps *perfSchema) StartStatement(sql string, connID uint64, callerName EnumC
 			return nil
 		}
 		source = fmt.Sprintf("%s:%d", fileName, fileLine)
+
+		callerLock.Lock()
```

Comment:
Consider the following two scenarios:
1. Two sessions check an empty callerNames with the same callerName at the same time. They will both write callerNames[callerName] to some source.
2. One session is writing to callerNames[callerName]. In the meanwhile, another session is reading from it.

Maybe scenario 2 will cause race problem. Use a ReadWriteLock will not cause performance issue.